pkg/envoy/ads/cache_callbacks.go

```diff
@@ -38,12 +38,6 @@ func (s *Server) OnStreamOpen(ctx context.Context, streamID int64, typ string) e
 
 	proxy := envoy.NewProxy(kind, uuid, si, utils.GetIPFromContext(ctx), streamID)
 
-	if err := s.recordPodMetadata(proxy); err == errServiceAccountMismatch {
-		// Service Account mismatch
-		log.Error().Err(err).Str("proxy", proxy.String()).Msg("Mismatched service account for proxy")
```

Comment:
Are we losing anything by not have the runtime check against the pods Service account?  dropping/moving the metadata struct to remove the dependency is good but is there a need to do this verification to ensure the cert matches what we expect?

Reply:
Ya I thought about that.. we can bring it back.

FWIW it's indirectly present in the GetPodForProxy call, but this is called on each iteration of building the configs and wouldn't return early as compared to here.

I'll leave the comment open to other's thoughts on if we want to keep it

Reply:
>FWIW it's indirectly present in the GetPodForProxy call, but this is called on each iteration of building the configs and wouldn't return early as compared to here.

ah now I remember it being there and wondering why it was in two places :-).  I like the explicit and early failure check here and I think it could still be platform independent.  let's see what other say

Reply:
re-added this back!